api/v1alpha4/kvmcluster_types.go

```diff
@@ -0,0 +1,80 @@
+/*
+Copyright 2021.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha4
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1alpha4"
+)
+
+// KVMClusterSpec defines the desired state of KVMCluster
+type KVMClusterSpec struct {
+	// ControlPlaneEndpoint represents the endpoint used to communicate with the control plane.
+	// +optional
+	ControlPlaneEndpoint clusterv1.APIEndpoint `json:"controlPlaneEndpoint"`
+}
+
+// KVMClusterStatus defines the observed state of KVMCluster
+type KVMClusterStatus struct {
```

Comment:
We need a `controlPlaneEndpoint` Field in the cluster status to comply with the CAPI provider contract. See https://cluster-api.sigs.k8s.io/developer/architecture/controllers/cluster.html#required-status-fields

Reply:
Same as above, I see that in the `spec`, not the `status`.